pandas/core/internals/blocks.py

```diff
@@ -3005,6 +3005,30 @@ def where(
             )
         return result
 
+    def replace(
+        self, to_replace, value, inplace=False, filter=None, regex=False, convert=True
+    ):
+        result = self if inplace else self.copy()
+        if filter is None:
```

Comment:
what case hits this?

Reply:
filter is none when replace is called on a Series as opposed to a DataFrame

Reply:
I still find this non-obvious.  can you add a comment to this effect

Reply:
i added a comment